### PR TITLE
[full-ci] - use KQL as default search query language

### PR DESCRIPTION
**Pull request by fschade**
## Description
Use [KQL](https://learn.microsoft.com/en-us/sharepoint/dev/general-development/keyword-query-language-kql-syntax-reference) as the default oCIS search query language.

Some examples of a valid KQL query are:

* `Tag`: `tag:golden tag:"silver"`
* `Filename`: `name:file.txt name:"file.docx"`
* `Content`: `content:ahab content:"captain aha*"`

Conjunctive normal form queries:

* `Boolean`: `tag:golden AND tag:"silver`, `tag:golden OR tag:"silver`, `tag:golden NOT tag:"silver`
* `Group`: `(tag:book content:ahab*)`, `tag:(book pdf)`

Complex queries:

* `(name:"moby di*" OR tag:bestseller) AND tag:book NOT tag:read`

## Related Issue
* https://github.com/owncloud/ocis/pull/7212
* https://github.com/owncloud/ocis/issues/7042
* https://github.com/owncloud/ocis/issues/7179
* https://github.com/owncloud/ocis/issues/7114
* https://github.com/owncloud/web/pull/9653
* https://github.com/owncloud/ocis/pull/7043
* https://github.com/owncloud/web/issues/9636
* https://github.com/owncloud/web/issues/9646

## Motivation and Context
Defining a own query language is error prone and leads to a lot of documentation, the oCIS team decided in that [ADR](https://github.com/owncloud/ocis/blob/docs/ocis/adr/0020-file-search-query-language.md) to use KQL as our standard. A clear set of rules is set by MS and the dictionary fits well in our usecase.

## How Has This Been Tested?
- unit tests added, have been adjusted
- acceptance tests have been adjusted 

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Technical debt

## Checklist:
- [X] Code changes
- [X] Unit tests added
- [ ] ~~Acceptance tests added~~
- [ ] ~~Documentation ticket raised: <link> ~~

### CODE REVIEW

File: tests/acceptance/features/apiSearch/apiSpaceSearch.feature [hunk 1]
```diff
@@ -20,7 +20,7 @@ Feature: Search
```
Comment (ScharfViktor) [inline tests/acceptance/features/apiSearch/apiSpaceSearch.feature:20]: ops, found one deficiency.
You have switched web to use space dav path, but tests still use only the new DAV path. we need to change it for all tests from apiSpaceSearch.feature like:

```
Scenario Outline: user can find data from the project space
    Given using <dav-path-version> DAV path
    When user "Alice" searches for "*fol*" using the WebDAV API
    Then the HTTP status code should be "207"
    And the search result should contain "4" entries
    And the search result of user "Alice" should contain these entries:
      | /folderMain                                           |
      | /folderMain/SubFolder1                                |
      | /folderMain/SubFolder1/subFOLDER2                     |
      | /folderMain/SubFolder1/subFOLDER2/insideTheFolder.txt |
    Examples:
      | dav-path-version |
      | old              |
      | new              |
    @skipOnRevaMaster
    Examples:
      | dav-path-version |
      | spaces           |
```

**Comment by fschade** [comment]
@2403905, @aduffeck i had to add dateTime support to the kql <-> bleve parser, compiler, and normalizer.

can you please have a look if it's ok for you, specially because of: 

https://github.com/owncloud/ocis/blob/cd7dd5643215de8908fb1677ed961da7d223bde5/services/search/pkg/search/service.go#L421

we should be ok now, i need another web bump because of:

https://github.com/owncloud/web/pull/9653

can you help @JanAckermann, @JammingBen

File: tests/acceptance/features/apiSearch/apiSpaceSearch.feature (hunk 1 above)
Comment (ScharfViktor) [reply tests/acceptance/features/apiSearch/apiSpaceSearch.feature:20]: > @ScharfViktor, not exactly sure what you mean, but if you mean that web uses the 'space/*' endpoint for report requests, ...

yes,  web uses the 'space/*' but api tests uses only new DAV Path. I changed it here https://github.com/owncloud/ocis/commit/109b70eea80a449a2bdad379fee93d937cf842c4 and I'll create PR
after merging your

**Review by aduffeck — COMMENTED** [review]
lgtm from reading the code but I haven't had a chance to test it yet.

**Review by kobergj — APPROVED** [review]
Awesome :+1:

**Comment by 2403905** [comment]
We also have issue with the  implicit logical operators 
`(tag:klass10) NOT tag:physik` will be transeted to tag:klass10 AND NOT tag:physik
But 
`tag:klass10 NOT tag:physik` will be transeted to tag:klass10 OR NOT tag:physik

**Comment by fschade** [comment]
@2403905 thats fine! the normalization could be improved, yes, but the current clients are only sending simple queries (i hope this changes soon ;) ), lets close (maerge) this here, improve our tests as a first step and then add the implementations step by step.... 

we first need to know (find out), what is really needed.

**Pull request merged**